### PR TITLE
SCANNPM-2 Implement running the Scanner Engine

### PR DESCRIPTION
**Pull request by 7PH**

### CODE REVIEW

File: src/java.ts [hunk 1]
```diff
@@ -132,12 +129,7 @@ export async function fetchJRE(
 
     await extractArchive(archivePath, jreDirPath);
 
-    const jreBinPath = path.join(jreDirPath, latestJREData.javaPath);
-
-    return {
-      ...latestJREData,
-      jrePath: jreBinPath,
-    };
```
Comment (7PH) [inline src/java.ts:140]: Other info arn't needed elsewhere, so I've changed this to just return the path to the binary

File: src/scanner-cli.ts [hunk 2]
```diff
@@ -77,13 +77,11 @@ function getScannerCliUrl(properties: ScannerProperties, version: string): URL {
 }
 
 export async function downloadScannerCli(properties: ScannerProperties): Promise<string> {
-  const token = properties[ScannerProperty.SonarToken];
   const version = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
   if (!/^[\d.]+$/.test(version)) {
     throw new Error(`Version "${version}" does not have a correct format."`);
   }
 
-  const proxyUrl = getProxyUrl(properties);
```
Comment (7PH) [inline src/scanner-cli.ts:86]: unrelated but cleaning up this unused var

File: src/java.ts (hunk 1 above)
Comment (lucas-paulger-sonarsource) [reply src/java.ts:140]: thanks for the callout

File: test/unit/scanner-engine.test.ts [hunk 3]
```diff
@@ -58,27 +67,14 @@ describe('scanner-engine', () => {
...
-      return Promise.resolve();
-    });
-
-    jest.spyOn(request, 'download').mockImplementation(() => {
-      return Promise.resolve();
-    });
+    jest.spyOn(file, 'getCachedFileLocation').mockResolvedValue(null);
+    jest.spyOn(file, 'extractArchive').mockResolvedValue();
+    jest.spyOn(file, 'validateChecksum').mockResolvedValue();
+    jest.spyOn(request, 'download').mockResolvedValue();
```
Comment (lucas-paulger-sonarsource) [inline test/unit/scanner-engine.test.ts:73]: much cleaner ty 👍🏻

File: src/scanner-engine.ts [hunk 4]
```diff
@@ -70,3 +72,55 @@ export async function fetchScannerEngine(properties: ScannerProperties) {
...
+    // Try and assume the log comes from the scanner engine
+    const parsed = JSON.parse(message) as ScannerLogEntry;
+    log(parsed.level, 'ScannerEngine', parsed.formattedMessage);
+    if (parsed.throwable) {
+      // Console.log without newline
+      process.stdout.write(parsed.throwable);
+    }
+  } catch (e) {
+    process.stdout.write(message);
+  }
```
[the remark this answers is not in the document]
Comment (7PH) [reply src/scanner-engine.ts:87]: Definitively, I'll do this as part of this PR

**Review by lucas-paulger-sonarsource — APPROVED** [review]
Tested, looks good. some logging improvements can be made but feel free to merge afterwards

File: src/scanner-cli.ts [hunk 5]
```diff
@@ -77,13 +77,11 @@ function getScannerCliUrl(properties: ScannerProperties, version: string): URL {
 }
 
 export async function downloadScannerCli(properties: ScannerProperties): Promise<string> {
-  const token = properties[ScannerProperty.SonarToken];
   const version = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
```
[the remark this answers is not in the document]
Comment (7PH) [reply src/scanner-cli.ts:80]: Totally makes sense to me!

**Pull request merged**